impl/spanner/directory/directory_test.go

```diff
@@ -0,0 +1,240 @@
+	t.Helper()
+	ddl, err := ktspanner.ReadDDL()
+	if err != nil {
+		t.Fatal(err)
+	}
+	client, cleanup := testutil.CreateDatabase(ctx, t, ddl)
+	s := New(ktspanner.NewForTest(client))
+	done := func(_ context.Context) { cleanup() }
+	return s, done
+}
+
+func TestList(t *testing.T) {
+	ctx := context.Background()
+	admin, cleanup := NewForTest(ctx, t)
+	defer cleanup(ctx)
+
+	directories := []*directory.Directory{
+		{
+			DirectoryID: "directory1",
+			Map:         &tpb.Tree{TreeId: 1},
+			Log:         &tpb.Tree{TreeId: 2},
+			VRF:         &keyspb.PublicKey{Der: []byte("pubkeybytes")},
+			VRFPriv:     &keyspb.PrivateKey{Der: []byte("privkeybytes")},
+			MinInterval: 1 * time.Second,
+			MaxInterval: 5 * time.Second,
+		},
+		{
+			DirectoryID: "directory2",
+			Map:         &tpb.Tree{TreeId: 1},
+			Log:         &tpb.Tree{TreeId: 2},
+			VRF:         &keyspb.PublicKey{Der: []byte("pubkeybytes")},
+			VRFPriv:     &keyspb.PrivateKey{Der: []byte("privkeybytes")},
+			MinInterval: 5 * time.Hour,
+			MaxInterval: 500 * time.Hour,
+			Deleted:     true,
+		},
+	}
+	for _, d := range directories {
+		if err := admin.Write(ctx, d); err != nil {
+			t.Errorf("Write(): %v", err)
+			continue
+		}
+		if err := admin.SetDelete(ctx, d.DirectoryID, d.Deleted); err != nil {
+			t.Errorf("SetDelete(%v, %v): %v", d.DirectoryID, d.Deleted, err)
+			continue
+		}
+	}
+
+	for _, tc := range []struct {
+		readDeleted bool
+		want        []*directory.Directory
+	}{
+		{
+			readDeleted: true,
+			want:        directories,
+		},
+		{
+			readDeleted: false,
+			want:        []*directory.Directory{directories[0]},
+		},
+	} {
+		directories, err := admin.List(ctx, tc.readDeleted)
+		if err != nil {
+			t.Errorf("List(): %v", err)
+			continue
+		}
+
+		if got, want := len(directories), len(tc.want); got != want {
+			t.Errorf("Got %v directories, want %v", got, want)
+		}
+		for i, d := range directories {
+			if got, want := d, tc.want[i]; !cmp.Equal(got, want, cmp.Comparer(proto.Equal)) {
+				t.Errorf("Directory[%v]: %v, want %v. Diff: %v", i, got, want, cmp.Diff(want, got))
+			}
+		}
+	}
+}
+
+func TestWriteReadDelete(t *testing.T) {
+	ctx := context.Background()
+	admin, cleanup := NewForTest(ctx, t)
+	defer cleanup(ctx)
+
+	d := &directory.Directory{
+		DirectoryID: "testdirectory",
+		Map:         &tpb.Tree{TreeId: 1},
+		Log:         &tpb.Tree{TreeId: 2},
+		VRF:         &keyspb.PublicKey{Der: []byte("pubkeybytes")},
+		VRFPriv:     &keyspb.PrivateKey{Der: []byte("privkeybytes")},
+		MinInterval: 1 * time.Second,
+		MaxInterval: 5 * time.Second,
+	}
+
+	for _, tc := range []struct {
+		desc                 string
+		write                bool
+		wantWriteErr         bool
+		setDelete, isDeleted bool
+		readDeleted          bool
+		wantReadCode         codes.Code
+	}{
+		{
+			desc:         "NotFound",
+			wantReadCode: codes.NotFound,
+		},
+		{
+			desc:  "Success",
+			write: true,
+		},
```

Comment:
These could be made 1-line literals, as they are very small. Potentially the next one as well.